documentation/specs/proposed/VS-OpenTelemetry.md

```diff
@@ -0,0 +1,138 @@
+# Telemetry via OpenTelemetry design
+
+VS OTel provide packages compatible with ingesting data to their backend if we instrument it via OpenTelemetry traces (System.Diagnostics.Activity).
+VS OTel packages are not open source so we need to conditionally include them in our build only for VS and MSBuild.exe
+
+[Onepager](https://github.com/dotnet/msbuild/blob/main/documentation/specs/proposed/telemetry-onepager.md)
+
+## Requirements
+
+### Performance
+
+- If not sampled, no infra initialization overhead.
+- Avoid allocations when not sampled.
+- Has to have no impact on Core without opting into tracing, small impact on Framework
+
+### Privacy
+
+- Hashing data points that could identify customers (e.g. names of targets)
+- Opt out capability
+
+### Security
+
+- Providing a method for creating a hook in Framework MSBuild
+- document the security implications of hooking custom telemetry Exporters/Collectors in Framework
```

Comment:
```suggestion
- If custom hooking solution will be used - document the security implications of hooking custom telemetry Exporters/Collectors in Framework
```

Since we plan to use AppDomainManager - we are using existing solution that is outside of our trust boundaries